ethereum_history_api/circuits/lib/README.md

```diff
@@ -1,58 +1,78 @@
-# Noir Ethereum history api circuits
+# Noir Ethereum history api library
 
-Noir circuits to generate account and storage proofs for historical blocks.
+Noir library for proving and verifying historical data on the Ethereum blockchain and other EVMs.
 
-## Compilation
+**Disclaimer:** This page is supposed to give you an idea about the usage & structure of the library, but it's still a good idea to consult the code for exact definitions of functions & types.
 
-To compile circuit files and install dependencies run:
+## Usage
 
-```sh
-nargo compile --workspace
-```
+There are two main ways to use this library. `With Oracles` or `Without Oracles`.
 
-To compile a single package run:
-```sh
-nargo compile --package package_name
-``` 
+### With oracles
 
-## Generating proofs
+If you decide to use Oracles - you don't need to provide all the data by hand and you can use functions that operate on a **high level of abstraction**. Those functions will then **fetch** required data such as raw header or state proofs from oracles, **verify** that the data is correct and return you the values that you've asked for.
 
-This circuit uses oracles and therefore you need to run the oracle server before generating proofs.
-Oracle server is located in [packages/noir-ethereum-api](../packages/noir-ethereum-api) and can be run by:
+Here is a list of public functions that you should use it you decide to go the `oracles route`:
```

Comment:
We need a paragraph about - that you need to run an oracle server, a single line command and a link to oracles project for more info